django/db/models/sql/datastructures.py

```diff
@@ -79,15 +82,36 @@ def as_sql(self, compiler, connection):
         qn = compiler.quote_name_unless_alias
         qn2 = connection.ops.quote_name
 
+        if hasattr(self.join_field, "get_joining_expressions"):
+            join_exprs = self.join_field.get_joining_expressions(
+                connection,
+                self.parent_alias,
+                self.table_alias,
+            )
+        else:
+            warnings.warn(
+                "The usage of get_joining_columns in Join is deprecated. Implement "
+                "get_joining_expressions instead.",
+                RemovedInDjango60Warning,
+            )
```

Comment:
This warning is not tested, but I'm not 100% certain it _has_ to stay 🤔 

Couldn't we rely on `get_joining_expressions`, and add to the release note that one should also probably override `get_joining_expressions` when they have overridden `get_joining_columns`?